src/pages/auth/login.tsx

```diff
@@ -29,22 +30,39 @@ const LoginPage = () => {
     setEmail(e.target.value);
   };
 
-  const handleSubmit = (e: React.ChangeEvent<HTMLInputElement>) => {
+  const handleSubmit = async (e: React.ChangeEvent<HTMLInputElement>) => {
     e.preventDefault();
+    setIsLoading(true);
 
-    if (email === testAcount.email) {
-      setEmailInvalid(false);
-    } else {
-      setEmailInvalid(true);
-    }
-    if (password === testAcount.password || password.length <= 5) {
-      setPasswordInvalid(false);
-    } else {
-      setPasswordInvalid(true);
-    }
-
-    if (email === testAcount.email && password === testAcount.password) {
-      alert(`Login maded with sucess. Welcome ${email}!!`);
+    try {
+      const response = await UserService.login({ email, password });
+      setIsLoading(false);
+      if (emailValidation(email) === false) {
+        setEmailInvalid(true);
+      }
+      if (passwordValidation(password) === false) {
+        setPasswordInvalid(true);
+        handleToast(
+          'Password must have min 5 and max 20 characters with 1 letter, 1 number and 1 special character',
+          'warning'
+        );
+      }
```

Comment:
These validations must be before the request.
Think that if the users doesn't fill an email or password correctly we shouldn't use resources.

Put it before the request and add a return false inside the ifs in order to stop the handleSubmit() when it fails, meaning that it won't continue unless the email and password are correct in the front-end side.

Reply:
It also can be `!passwordValidation(password)` instead of `passwordValidation(password) === false`